samples/sensor/mcux_lpcmp/src/main.c

```diff
@@ -0,0 +1,123 @@
+/*
+ * Copyright NXP 2024
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr/device.h>
+#include <zephyr/drivers/sensor.h>
+#include <zephyr/drivers/sensor/mcux_lpcmp.h>
+#include <zephyr/kernel.h>
+#include <stdio.h>
+
+struct lpcmp_attr {
+	int16_t attr;
+	int32_t val;
+};
+
+static const struct lpcmp_attr attrs[] = {
+	{.attr = SENSOR_ATTR_MCUX_LPCMP_POSITIVE_MUX_INPUT, .val = 0},
+	{.attr = SENSOR_ATTR_MCUX_LPCMP_NEGATIVE_MUX_INPUT, .val = 7},
+
+	{.attr = SENSOR_ATTR_MCUX_LPCMP_DAC_ENABLE, .val = 1},
+	{.attr = SENSOR_ATTR_MCUX_LPCMP_DAC_HIGH_POWER_MODE_ENABLE, .val = 0},
+	{.attr = SENSOR_ATTR_MCUX_LPCMP_DAC_REFERENCE_VOLTAGE_SOURCE, .val = 0},
+	{.attr = SENSOR_ATTR_MCUX_LPCMP_DAC_OUTPUT_VOLTAGE, .val = (0xFF >> 1)},
+
+	{.attr = SENSOR_ATTR_MCUX_LPCMP_SAMPLE_ENABLE, .val = 0},
+	{.attr = SENSOR_ATTR_MCUX_LPCMP_FILTER_COUNT, .val = 7},
+	{.attr = SENSOR_ATTR_MCUX_LPCMP_FILTER_PERIOD, .val = (0xFF >> 1)},
+
+	{.attr = SENSOR_ATTR_MCUX_LPCMP_COUTA_WINDOW_ENABLE, .val = 0},
+	{.attr = SENSOR_ATTR_MCUX_LPCMP_COUTA_WINDOW_SIGNAL_INVERT_ENABLE, .val = 0},
+	{.attr = SENSOR_ATTR_MCUX_LPCMP_COUTA_SIGNAL, .val = 0},
+	{.attr = SENSOR_ATTR_MCUX_LPCMP_COUT_EVENT_TO_CLOSE_WINDOW, .val = 0},
+};
+
+#ifdef CONFIG_MCUX_LPCMP_TRIGGER
+static const int16_t triggers[] = {
+	SENSOR_TRIG_MCUX_LPCMP_OUTPUT_RISING,
+	SENSOR_TRIG_MCUX_LPCMP_OUTPUT_FALLING,
+};
+#endif /* CONFIG_MCUX_LPCMP_TRIGGER */
+
+static void lpcmp_input_handler(bool above_threshold)
+{
+	if (above_threshold) {
+		printf("LPCMP input above threshold\r\n");
+	} else {
+		printf("LPCMP input below threshold\r\n");
```

Comment:
No need for the `\r` here, Zephyr will handle this internally when outputting via printf to a console